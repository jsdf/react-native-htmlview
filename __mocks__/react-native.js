var React = require.requireActual('react')

var ReactNative = Object.assign({}, React, {
  ListView: jest.genMockFromModule('./View'),
  View: jest.genMockFromModule('./View'),
  Text: jest.genMockFromModule('./View'),
  StyleSheet: {
    create: (ss) => ss,
  },
  PropTypes: React.PropTypes,
  Linking: {
    openURL: jest.fn(),
  },
  Dimensions: {
    get: function() {
      return {width: 414, height: 736}
    },
  },
})

module.exports = ReactNative
