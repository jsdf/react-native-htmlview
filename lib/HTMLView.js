'use strict';Object.defineProperty(exports,"__esModule",{value:true});var _jsxFileName='src/HTMLView.js';var _extends=Object.assign||function(target){for(var i=1;i<arguments.length;i++){var source=arguments[i];for(var key in source){if(Object.prototype.hasOwnProperty.call(source,key)){target[key]=source[key];}}}return target;};var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _react=require('react');var _react2=_interopRequireDefault(_react);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _htmlToElement=require('./htmlToElement');var _htmlToElement2=_interopRequireDefault(_htmlToElement);
var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}

var boldStyle={fontWeight:'500'};
var italicStyle={fontStyle:'italic'};
var underlineStyle={textDecorationLine:'underline'};
var strikethroughStyle={textDecorationLine:'line-through'};
var codeStyle={fontFamily:_reactNative.Platform.OS==='ios'?'Menlo':'monospace'};

var baseStyles=_reactNative.StyleSheet.create({
b:boldStyle,
strong:boldStyle,
i:italicStyle,
em:italicStyle,
u:underlineStyle,
s:strikethroughStyle,
strike:strikethroughStyle,
pre:codeStyle,
code:codeStyle,
a:{
fontWeight:'500',
color:'#007AFF'},

h1:{fontWeight:'500',fontSize:36},
h2:{fontWeight:'500',fontSize:30},
h3:{fontWeight:'500',fontSize:24},
h4:{fontWeight:'500',fontSize:18},
h5:{fontWeight:'500',fontSize:14},
h6:{fontWeight:'500',fontSize:12}});


var htmlToElementOptKeys=[
'lineBreak',
'paragraphBreak',
'bullet',
'TextComponent',
'textComponentProps',
'NodeComponent',
'nodeComponentProps'];var


HtmlView=function(_PureComponent){_inherits(HtmlView,_PureComponent);
function HtmlView(){_classCallCheck(this,HtmlView);var _this=_possibleConstructorReturn(this,(HtmlView.__proto__||Object.getPrototypeOf(HtmlView)).call(this));

_this.state={
element:null};return _this;

}_createClass(HtmlView,[{key:'componentDidMount',value:function componentDidMount()

{
this.mounted=true;
this.startHtmlRender(this.props.value);
}},{key:'componentWillReceiveProps',value:function componentWillReceiveProps(

nextProps){
if(
this.props.value!==nextProps.value||
this.props.stylesheet!==nextProps.stylesheet||
this.props.textComponentProps!==nextProps.textComponentProps)
{
this.startHtmlRender(
nextProps.value,
nextProps.stylesheet,
nextProps.textComponentProps);

}
}},{key:'componentWillUnmount',value:function componentWillUnmount()

{
this.mounted=false;
}},{key:'startHtmlRender',value:function startHtmlRender(

value,style,textComponentProps){var _this2=this;var _props=







this.props,addLineBreaks=_props.addLineBreaks,onLinkPress=_props.onLinkPress,onLinkLongPress=_props.onLinkLongPress,stylesheet=_props.stylesheet,renderNode=_props.renderNode,onError=_props.onError;

if(!value){
this.setState({element:null});
}

var opts={
addLineBreaks:addLineBreaks,
linkHandler:onLinkPress,
linkLongPressHandler:onLinkLongPress,
styles:_extends({},baseStyles,stylesheet,style),
customRenderer:renderNode};


htmlToElementOptKeys.forEach(function(key){
if(typeof _this2.props[key]!=='undefined'){
opts[key]=_this2.props[key];
}
});

if(textComponentProps){
opts.textComponentProps=textComponentProps;
}

(0,_htmlToElement2.default)(value,opts,function(err,element){
if(err){
onError(err);
}

if(_this2.mounted){
_this2.setState({element:element});
}
});
}},{key:'render',value:function render()

{var _props2=
this.props,RootComponent=_props2.RootComponent,style=_props2.style;var
element=this.state.element;
if(element){
return(
_react2.default.createElement(RootComponent,_extends({},this.props.rootComponentProps,{style:style,__source:{fileName:_jsxFileName,lineNumber:123}}),
element));


}
return _react2.default.createElement(RootComponent,_extends({},this.props.rootComponentProps,{style:style,__source:{fileName:_jsxFileName,lineNumber:128}}));
}}]);return HtmlView;}(_react.PureComponent);


HtmlView.propTypes={
addLineBreaks:_propTypes2.default.bool,
bullet:_propTypes2.default.string,
lineBreak:_propTypes2.default.string,
NodeComponent:_propTypes2.default.func,
nodeComponentProps:_propTypes2.default.object,
onError:_propTypes2.default.func,
onLinkPress:_propTypes2.default.func,
onLinkLongPress:_propTypes2.default.func,
paragraphBreak:_propTypes2.default.string,
renderNode:_propTypes2.default.func,
RootComponent:_propTypes2.default.func,
rootComponentProps:_propTypes2.default.object,
style:_reactNative.ViewPropTypes.style,
stylesheet:_propTypes2.default.object,
TextComponent:_propTypes2.default.func,
textComponentProps:_propTypes2.default.object,
value:_propTypes2.default.string};


HtmlView.defaultProps={
addLineBreaks:true,
onLinkPress:function onLinkPress(url){return _reactNative.Linking.openURL(url);},
onLinkLongPress:null,
onError:console.error.bind(console),
RootComponent:function RootComponent(element){return _react2.default.createElement(_reactNative.View,_extends({},element,{__source:{fileName:_jsxFileName,lineNumber:157}}));}};exports.default=


HtmlView;
//# sourceMappingURL=HTMLView.js.map