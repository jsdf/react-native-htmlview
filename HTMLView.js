import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import htmlToElement from './htmlToElement';
import {Linking, Platform, StyleSheet, View, ViewPropTypes} from 'react-native';

const boldStyle = {fontWeight: 'bold'};
const italicStyle = {fontStyle: 'italic'};
const underlineStyle = {textDecorationLine: 'underline'};
const strikethroughStyle = {textDecorationLine: 'line-through'};
const codeStyle = {fontFamily: Platform.OS === 'ios' ? 'Menlo' : 'monospace'};

const baseStyles = StyleSheet.create({
  b: boldStyle,
  strong: boldStyle,
  i: italicStyle,
  em: italicStyle,
  u: underlineStyle,
  s: strikethroughStyle,
  strike: strikethroughStyle,
  pre: codeStyle,
  code: codeStyle,
  a: {
    fontWeight: 'bold',
    color: '#007AFF',
  },
  h1: {fontWeight: 'bold', fontSize: 36},
  h2: {fontWeight: 'bold', fontSize: 30},
  h3: {fontWeight: 'bold', fontSize: 24},
  h4: {fontWeight: 'bold', fontSize: 18},
  h5: {fontWeight: 'bold', fontSize: 14},
  h6: {fontWeight: 'bold', fontSize: 12},
});

const htmlToElementOptKeys = [
  'lineBreak',
  'paragraphBreak',
  'bullet',
  'TextComponent',
  'textComponentProps',
  'NodeComponent',
  'nodeComponentProps',
];

class HtmlView extends PureComponent {
  constructor() {
    super();
    this.state = {
      element: null,
    };
  }

  componentDidMount() {
    this.mounted = true;
    this.startHtmlRender(this.props.value);
  }

  componentDidUpdate(prevProps) {
    if (this.props.value !== prevProps.value || this.props.stylesheet !== prevProps.stylesheet || this.props.textComponentProps !== prevProps.textComponentProps || this.props.nodeComponentProps !== prevProps.nodeComponentProps) {
      this.startHtmlRender(this.props.value, this.props.stylesheet, this.props.textComponentProps, this.props.nodeComponentProps);
    }
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  startHtmlRender(value, style, textComponentProps, nodeComponentProps) {
    const {
      addLineBreaks,
      onLinkPress,
      onLinkLongPress,
      stylesheet,
      renderNode,
      onError,
    } = this.props;

    if (!value) {
      this.setState({element: null});
    }

    const opts = {
      addLineBreaks,
      linkHandler: onLinkPress,
      linkLongPressHandler: onLinkLongPress,
      styles: {...baseStyles, ...stylesheet, ...style},
      customRenderer: renderNode,
    };

    htmlToElementOptKeys.forEach(key => {
      if (typeof this.props[key] !== 'undefined') {
        opts[key] = this.props[key];
      }
    });

    if (textComponentProps) {
      opts.textComponentProps = textComponentProps;
    }

    if (nodeComponentProps) {
      opts.nodeComponentProps = nodeComponentProps;
    }

    htmlToElement(value, opts, (err, element) => {
      if (err) {
        onError(err);
      }

      if (this.mounted) {
        this.setState({element});
      }
    });
  }

  render() {
    const {RootComponent, style} = this.props;
    const {element} = this.state;
    if (element) {
      return (
        <RootComponent
          {...this.props.rootComponentProps}
          style={style}
        >
          {element}
        </RootComponent>
      );
    }
    return (
      <RootComponent
        {...this.props.rootComponentProps}
        style={style}
      />
    );
  }
}

HtmlView.propTypes = {
  addLineBreaks: PropTypes.bool,
  bullet: PropTypes.string,
  lineBreak: PropTypes.string,
  NodeComponent: PropTypes.func,
  nodeComponentProps: PropTypes.object,
  onError: PropTypes.func,
  onLinkPress: PropTypes.func,
  onLinkLongPress: PropTypes.func,
  paragraphBreak: PropTypes.string,
  renderNode: PropTypes.func,
  RootComponent: PropTypes.func,
  rootComponentProps: PropTypes.object,
  style: ViewPropTypes.style,
  stylesheet: PropTypes.object,
  TextComponent: PropTypes.func,
  textComponentProps: PropTypes.object,
  value: PropTypes.string,
};

HtmlView.defaultProps = {
  addLineBreaks: true,
  onLinkPress: url => Linking.openURL(url),
  onLinkLongPress: null,
  onError: console.error.bind(console),
  RootComponent: element => <View {...element} />, // eslint-disable-line react/display-name
};

export default HtmlView;
