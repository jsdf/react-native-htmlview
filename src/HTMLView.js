import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import htmlToElement from './htmlToElement';
import {Linking, Platform, StyleSheet, View, ViewPropTypes} from 'react-native';

const boldStyle = {fontWeight: '500'};
const italicStyle = {fontStyle: 'italic'};
const underlineStyle = {textDecorationLine: 'underline'};
const strikethroughStyle = {textDecorationLine: 'line-through'};
const codeStyle = {fontFamily: Platform.OS === 'ios' ? 'Menlo' : 'monospace'};

const baseStyles = StyleSheet.create({
  b: boldStyle,
  strong: boldStyle,
  i: italicStyle,
  em: italicStyle,
  u: underlineStyle,
  s: strikethroughStyle,
  strike: strikethroughStyle,
  pre: codeStyle,
  code: codeStyle,
  a: {
    fontWeight: '500',
    color: '#007AFF',
  },
  h1: {fontWeight: '500', fontSize: 36},
  h2: {fontWeight: '500', fontSize: 30},
  h3: {fontWeight: '500', fontSize: 24},
  h4: {fontWeight: '500', fontSize: 18},
  h5: {fontWeight: '500', fontSize: 14},
  h6: {fontWeight: '500', fontSize: 12},
});

const htmlToElementOptKeys = [
  'lineBreak',
  'paragraphBreak',
  'bullet',
  'TextComponent',
  'textComponentProps',
  'NodeComponent',
  'nodeComponentProps',
];

class HtmlView extends PureComponent {
  constructor() {
    super();
    this.state = {
      element: null,
    };
  }

  componentDidMount() {
    this.mounted = true;
    this.startHtmlRender(this.props.value);
  }

  componentWillReceiveProps(nextProps) {
    if (
      this.props.value !== nextProps.value ||
      this.props.stylesheet !== nextProps.stylesheet ||
      this.props.textComponentProps !== nextProps.textComponentProps
    ) {
      this.startHtmlRender(
        nextProps.value,
        nextProps.stylesheet,
        nextProps.textComponentProps
      );
    }
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  startHtmlRender(value, style, textComponentProps) {
    const {
      addLineBreaks,
      onLinkPress,
      onLinkLongPress,
      stylesheet,
      renderNode,
      onError,
    } = this.props;

    if (!value) {
      this.setState({element: null});
    }

    const opts = {
      addLineBreaks,
      linkHandler: onLinkPress,
      linkLongPressHandler: onLinkLongPress,
      styles: {...baseStyles, ...stylesheet, ...style},
      customRenderer: renderNode,
    };

    htmlToElementOptKeys.forEach(key => {
      if (typeof this.props[key] !== 'undefined') {
        opts[key] = this.props[key];
      }
    });

    if (textComponentProps) {
      opts.textComponentProps = textComponentProps;
    }

    htmlToElement(value, opts, (err, element) => {
      if (err) {
        onError(err);
      }

      if (this.mounted) {
        this.setState({element});
      }
    });
  }

  render() {
    const {RootComponent, style} = this.props;
    const {element} = this.state;
    if (element) {
      return (
        <RootComponent {...this.props.rootComponentProps} style={style}>
          {element}
        </RootComponent>
      );
    }
    return <RootComponent {...this.props.rootComponentProps} style={style} />;
  }
}

HtmlView.propTypes = {
  addLineBreaks: PropTypes.bool,
  bullet: PropTypes.string,
  lineBreak: PropTypes.string,
  NodeComponent: PropTypes.func,
  nodeComponentProps: PropTypes.object,
  onError: PropTypes.func,
  onLinkPress: PropTypes.func,
  onLinkLongPress: PropTypes.func,
  paragraphBreak: PropTypes.string,
  renderNode: PropTypes.func,
  RootComponent: PropTypes.func,
  rootComponentProps: PropTypes.object,
  style: ViewPropTypes.style,
  stylesheet: PropTypes.object,
  TextComponent: PropTypes.func,
  textComponentProps: PropTypes.object,
  value: PropTypes.string,
};

HtmlView.defaultProps = {
  addLineBreaks: true,
  onLinkPress: url => Linking.openURL(url),
  onLinkLongPress: null,
  onError: console.error.bind(console),
  RootComponent: element => <View {...element} />, // eslint-disable-line react/display-name
};

export default HtmlView;
