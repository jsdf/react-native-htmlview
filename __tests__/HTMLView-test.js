import React from 'react';

import renderer from 'react-test-renderer';

import {StyleSheet, Text} from 'react-native';

import HTMLView from '../HTMLView';

describe('<HTMLView/>', () => {
  it('should render a <Text/> element', () => {
    const htmlContent = '<p><a href="http://jsdf.co">&hearts nice job!</a></p>';

    expect(
      renderer.create(<HTMLView value={htmlContent} />).toJSON()
    ).toMatchSnapshot();
  });

  it('should render text in a (nested) <Text/> element', () => {
    const htmlContent = 'This is some text';

    expect(
      renderer.create(<HTMLView value={htmlContent} />).toJSON()
    ).toMatchSnapshot();
  });

  it('should render ul bullets', () => {
    const htmlContent = '<ul><li> a </li><li> b </li></ul>';

    expect(
      renderer.create(<HTMLView value={htmlContent} />).toJSON()
    ).toMatchSnapshot();
  });

  it('should render ol numbers', () => {
    const htmlContent = '<ol><li> a </li><li> b </li></ol>';

    expect(
      renderer.create(<HTMLView value={htmlContent} />).toJSON()
    ).toMatchSnapshot();
  });

  it('should handle additional text nodes between list items', () => {
    const htmlContent = `<ol>
      <li> a </li>
      <li> b </li>
    </ol>`;

    expect(
      renderer.create(<HTMLView value={htmlContent} />).toJSON()
    ).toMatchSnapshot();
  });

  it('should render an <Image />, with default width/height of 1', () => {
    const imgSrc =
      'https://facebook.github.io/react-native/img/header_logo.png';
    const htmlContent = `<img src="${imgSrc}"/>`;

    expect(
      renderer.create(<HTMLView value={htmlContent} />).toJSON()
    ).toMatchSnapshot();
  });

  it('should render an <Image /> with set width/height', () => {
    const imgSrc =
      'https://facebook.github.io/react-native/img/header_logo.png';
    const htmlContent = `<img src="${imgSrc}" width="66" height="58"/>`;

    expect(
      renderer.create(<HTMLView value={htmlContent} />).toJSON()
    ).toMatchSnapshot();
  });

  it('should render inherited styles correctly', () => {
    const htmlContent = '<b>RED<u>BLUE<i>GREEN</i></u></b>';
    const stylesheet = StyleSheet.create({
      b: {color: 'red'},
      u: {color: 'blue'},
      i: {color: 'green'},
    });

    expect(
      renderer
        .create(<HTMLView value={htmlContent} stylesheet={stylesheet} />)
        .toJSON()
    ).toMatchSnapshot();
  });

  it('should render shoddy html including headings, links, bold, italic', () => {
    const htmlContent = `
      <div class="comment">
        <span class="c00">
          <h2>&gt; Dwayne’s only companion at night was a Labrador retriever named Sparky.</h2>
        <p>
        <i>Sparky could not wag his tail-because of an automobile accident many years ago, so he had no way of telling other dogs how friendly he was.
        He opened the door of the cage, something Bill couldn’t have done in a thousand years. Bill flew over to a windowsill.
        <b>The undippable flag was a beauty, and the anthem and the vacant motto might not have mattered much, if it weren’t for this: a lot of citizens were so ignored and cheated and insulted that they thought they might be in the wrong country, or even on the wrong planet, that some terrible mistake had been made.
        </p>
        <p>
          [1] <a href="https://code.facebook.com/posts/1505962329687926/flow-a-new-static-type-checker-for-javascript/" rel="nofollow">https://code.facebook.com/posts/1505962329687926/flow-a-new-...</a>
        </p>
      </span>
    </div>
    `;

    expect(
      renderer.create(<HTMLView value={htmlContent} />).toJSON()
    ).toMatchSnapshot();
  });

  it('should not render extra linebreaks if configured not to', () => {
    const htmlContent = `
      <div>
          <h2>Dwayne’s only companion at night was a Labrador retriever named Sparky.</h2>
        <p>
        Sparky could not wag his tail-because of an automobile accident many years ago, so he had no way of telling other dogs how friendly he was.<br>
        He opened the door of the cage, something Bill couldn’t have done in a thousand years. Bill flew over to a windowsill.
        </p>
      </div>
    `;

    expect(
      renderer
        .create(<HTMLView value={htmlContent} addLineBreaks={false} />)
        .toJSON()
    ).toMatchSnapshot();
  });

  it('should not render extra linebreaks in list items if configured not to', () => {
    const htmlContent = '<ul><li> a </li><li> b </li></ul>';

    expect(
      renderer
        .create(<HTMLView value={htmlContent} addLineBreaks={false} />)
        .toJSON()
    ).toMatchSnapshot();
  });

  it('can use a custom renderer', () => {
    const htmlContent = `
      <div>
        <thing a="b" />
    </div>
    `;

    function renderNode(node, index) {
      if (node.name == 'thing') {
        return <Text key={index}>{node.attribs.b}</Text>;
      }
    }

    expect(
      renderer
        .create(<HTMLView value={htmlContent} renderNode={renderNode} />)
        .toJSON()
    ).toMatchSnapshot();
  });

  it('can use a custom node class', () => {
    class Node extends React.Component {
      render() {
        return <Text {...this.props} selectable={false} />;
      }
    }

    const htmlContent = `
      <div>
        <div a="b" />
      </div>
    `;

    expect(
      renderer
        .create(<HTMLView value={htmlContent} NodeComponent={Node} />)
        .toJSON()
    ).toMatchSnapshot();
  });

  it('can use custom node props', () => {
    const htmlContent = `
      <div>
        <div a="b" />
      </div>
    `;

    expect(
      renderer
        .create(
          <HTMLView
            value={htmlContent}
            nodeComponentProps={{selectable: false}}
          />
        )
        .toJSON()
    ).toMatchSnapshot();
  });
});
