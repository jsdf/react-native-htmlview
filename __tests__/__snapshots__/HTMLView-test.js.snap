// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HTMLView/> can use a custom node class 1`] = `
<View>
  <Text
    style={
      Array [
        null,
        null,
      ]
    }
  >
    
      
  </Text>
  <Text
    onLongPress={null}
    onPress={null}
    selectable={false}
  >
    <Text
      style={
        Array [
          null,
          Object {},
        ]
      }
    >
      
        
    </Text>
    <Text
      onLongPress={null}
      onPress={null}
      selectable={false}
      style={null}
    >
      <Text
        style={
          Array [
            null,
            Object {},
          ]
        }
      >
        
      
      </Text>
    </Text>
    <Text
      style={
        Array [
          null,
          Object {},
        ]
      }
    >
      
    
    </Text>
  </Text>
</View>
`;

exports[`<HTMLView/> can use a custom renderer 1`] = `
<View>
  <Text
    style={
      Array [
        null,
        null,
      ]
    }
  >
    
      
  </Text>
  <Text
    onLongPress={null}
    onPress={null}
  >
    <Text
      style={
        Array [
          null,
          Object {},
        ]
      }
    >
      
        
    </Text>
    <Text />
  </Text>
  <Text
    style={
      Array [
        null,
        null,
      ]
    }
  >
    
    
  </Text>
</View>
`;

exports[`<HTMLView/> can use custom node props 1`] = `
<View>
  <Text
    style={
      Array [
        null,
        null,
      ]
    }
  >
    
      
  </Text>
  <Text
    onLongPress={null}
    onPress={null}
    selectable={false}
  >
    <Text
      style={
        Array [
          null,
          Object {},
        ]
      }
    >
      
        
    </Text>
    <Text
      onLongPress={null}
      onPress={null}
      selectable={false}
      style={null}
    >
      <Text
        style={
          Array [
            null,
            Object {},
          ]
        }
      >
        
      
      </Text>
    </Text>
    <Text
      style={
        Array [
          null,
          Object {},
        ]
      }
    >
      
    
    </Text>
  </Text>
</View>
`;

exports[`<HTMLView/> should handle additional text nodes between list items 1`] = `
<View>
  <Text
    onLongPress={null}
    onPress={null}
  >
    <Text
      style={
        Array [
          null,
          Object {},
        ]
      }
    >
      
      
    </Text>
    <Text
      onLongPress={null}
      onPress={null}
      style={null}
    >
      <Text
        style={
          Array [
            null,
            Object {},
          ]
        }
      >
        1. 
      </Text>
      <Text
        style={
          Array [
            null,
            Object {},
          ]
        }
      >
         a 
      </Text>
      

    </Text>
    <Text
      style={
        Array [
          null,
          Object {},
        ]
      }
    >
      
      
    </Text>
    <Text
      onLongPress={null}
      onPress={null}
      style={null}
    >
      <Text
        style={
          Array [
            null,
            Object {},
          ]
        }
      >
        2. 
      </Text>
      <Text
        style={
          Array [
            null,
            Object {},
          ]
        }
      >
         b 
      </Text>
      

    </Text>
    <Text
      style={
        Array [
          null,
          Object {},
        ]
      }
    >
      
    
    </Text>
  </Text>
</View>
`;

exports[`<HTMLView/> should not render extra linebreaks if configured not to 1`] = `
<View>
  <Text
    style={
      Array [
        null,
        null,
      ]
    }
  >
    
      
  </Text>
  <Text
    onLongPress={null}
    onPress={null}
  >
    <Text
      style={
        Array [
          null,
          Object {},
        ]
      }
    >
      
          
    </Text>
    <Text
      onLongPress={null}
      onPress={null}
      style={null}
    >
      <Text
        style={
          Array [
            null,
            Object {
              "fontSize": 30,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Dwayne’s only companion at night was a Labrador retriever named Sparky.
      </Text>
    </Text>
    <Text
      style={
        Array [
          null,
          Object {},
        ]
      }
    >
      
        
    </Text>
    <Text
      onLongPress={null}
      onPress={null}
      style={null}
    >
      <Text
        style={
          Array [
            null,
            Object {},
          ]
        }
      >
        
        Sparky could not wag his tail-because of an automobile accident many years ago, so he had no way of telling other dogs how friendly he was.
      </Text>
      <Text
        onLongPress={null}
        onPress={null}
        style={null}
      />
      <Text
        style={
          Array [
            null,
            Object {},
          ]
        }
      >
        
        He opened the door of the cage, something Bill couldn’t have done in a thousand years. Bill flew over to a windowsill.
        
      </Text>
    </Text>
    <Text
      style={
        Array [
          null,
          Object {},
        ]
      }
    >
      
      
    </Text>
  </Text>
  <Text
    style={
      Array [
        null,
        null,
      ]
    }
  >
    
    
  </Text>
</View>
`;

exports[`<HTMLView/> should not render extra linebreaks in list items if configured not to 1`] = `
<View>
  <Text
    onLongPress={null}
    onPress={null}
  >
    <Text
      onLongPress={null}
      onPress={null}
      style={null}
    >
      <Text
        style={
          Array [
            null,
            Object {},
          ]
        }
      >
        • 
      </Text>
      <Text
        style={
          Array [
            null,
            Object {},
          ]
        }
      >
         a 
      </Text>
    </Text>
    <Text
      onLongPress={null}
      onPress={null}
      style={null}
    >
      <Text
        style={
          Array [
            null,
            Object {},
          ]
        }
      >
        • 
      </Text>
      <Text
        style={
          Array [
            null,
            Object {},
          ]
        }
      >
         b 
      </Text>
    </Text>
  </Text>
</View>
`;

exports[`<HTMLView/> should render a <Text/> element 1`] = `
<View>
  <Text
    onLongPress={null}
    onPress={null}
  >
    <Text
      onLongPress={null}
      onPress={[Function]}
      style={null}
    >
      <Text
        style={
          Array [
            null,
            Object {
              "color": "#007AFF",
              "fontWeight": "bold",
            },
          ]
        }
      >
        &hearts nice job!
      </Text>
    </Text>
  </Text>
</View>
`;

exports[`<HTMLView/> should render an <Image /> with set width/height 1`] = `
<View>
  <Image
    source={
      Object {
        "height": 58,
        "uri": "https://facebook.github.io/react-native/img/header_logo.png",
        "width": 66,
      }
    }
    style={
      Object {
        "backgroundColor": "transparent",
        "height": 58,
        "width": 66,
      }
    }
  />
</View>
`;

exports[`<HTMLView/> should render an <Image />, with default width/height of 1 1`] = `
<View>
  <Image
    source={
      Object {
        "height": 1,
        "uri": "https://facebook.github.io/react-native/img/header_logo.png",
        "width": 1,
      }
    }
    style={
      Object {
        "backgroundColor": "transparent",
        "height": 1,
        "width": 1,
      }
    }
  />
</View>
`;

exports[`<HTMLView/> should render inherited styles correctly 1`] = `
<View>
  <Text
    onLongPress={null}
    onPress={null}
    style={
      Object {
        "color": "red",
      }
    }
  >
    <Text
      style={
        Array [
          null,
          Object {
            "color": "red",
          },
        ]
      }
    >
      RED
    </Text>
    <Text
      onLongPress={null}
      onPress={null}
      style={null}
    >
      <Text
        style={
          Array [
            null,
            Object {
              "color": "blue",
            },
          ]
        }
      >
        BLUE
      </Text>
      <Text
        onLongPress={null}
        onPress={null}
        style={null}
      >
        <Text
          style={
            Array [
              null,
              Object {
                "color": "green",
              },
            ]
          }
        >
          GREEN
        </Text>
      </Text>
    </Text>
  </Text>
</View>
`;

exports[`<HTMLView/> should render ol numbers 1`] = `
<View>
  <Text
    onLongPress={null}
    onPress={null}
  >
    <Text
      onLongPress={null}
      onPress={null}
      style={null}
    >
      <Text
        style={
          Array [
            null,
            Object {},
          ]
        }
      >
        1. 
      </Text>
      <Text
        style={
          Array [
            null,
            Object {},
          ]
        }
      >
         a 
      </Text>
      

    </Text>
    <Text
      onLongPress={null}
      onPress={null}
      style={null}
    >
      <Text
        style={
          Array [
            null,
            Object {},
          ]
        }
      >
        2. 
      </Text>
      <Text
        style={
          Array [
            null,
            Object {},
          ]
        }
      >
         b 
      </Text>
    </Text>
  </Text>
</View>
`;

exports[`<HTMLView/> should render shoddy html including headings, links, bold, italic 1`] = `
<View>
  <Text
    style={
      Array [
        null,
        null,
      ]
    }
  >
    
      
  </Text>
  <Text
    onLongPress={null}
    onPress={null}
  >
    <Text
      style={
        Array [
          null,
          Object {},
        ]
      }
    >
      
        
    </Text>
    <Text
      onLongPress={null}
      onPress={null}
      style={null}
    >
      <Text
        style={
          Array [
            null,
            Object {},
          ]
        }
      >
        
          
      </Text>
      <Text
        onLongPress={null}
        onPress={null}
        style={null}
      >
        <Text
          style={
            Array [
              null,
              Object {
                "fontSize": 30,
                "fontWeight": "bold",
              },
            ]
          }
        >
          &gt; Dwayne’s only companion at night was a Labrador retriever named Sparky.
        </Text>
        

      </Text>
      <Text
        style={
          Array [
            null,
            Object {},
          ]
        }
      >
        
        
      </Text>
      <Text
        onLongPress={null}
        onPress={null}
        style={null}
      >
        <Text
          style={
            Array [
              null,
              Object {},
            ]
          }
        >
          
        
        </Text>
        <Text
          onLongPress={null}
          onPress={null}
          style={null}
        >
          <Text
            style={
              Array [
                null,
                Object {
                  "fontStyle": "italic",
                },
              ]
            }
          >
            Sparky could not wag his tail-because of an automobile accident many years ago, so he had no way of telling other dogs how friendly he was.
        He opened the door of the cage, something Bill couldn’t have done in a thousand years. Bill flew over to a windowsill.
        
          </Text>
          <Text
            onLongPress={null}
            onPress={null}
            style={null}
          >
            <Text
              style={
                Array [
                  null,
                  Object {
                    "fontStyle": "italic",
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              The undippable flag was a beauty, and the anthem and the vacant motto might not have mattered much, if it weren’t for this: a lot of citizens were so ignored and cheated and insulted that they thought they might be in the wrong country, or even on the wrong planet, that some terrible mistake had been made.
        
            </Text>
          </Text>
        </Text>
        


      </Text>
      <Text
        style={
          Array [
            null,
            Object {},
          ]
        }
      >
        
        
      </Text>
      <Text
        onLongPress={null}
        onPress={null}
        style={null}
      >
        <Text
          style={
            Array [
              null,
              Object {},
            ]
          }
        >
          
          [1] 
        </Text>
        <Text
          onLongPress={null}
          onPress={[Function]}
          style={null}
        >
          <Text
            style={
              Array [
                null,
                Object {
                  "color": "#007AFF",
                  "fontWeight": "bold",
                },
              ]
            }
          >
            https://code.facebook.com/posts/1505962329687926/flow-a-new-...
          </Text>
        </Text>
        <Text
          style={
            Array [
              null,
              Object {},
            ]
          }
        >
          
        
        </Text>
        


      </Text>
      <Text
        style={
          Array [
            null,
            Object {},
          ]
        }
      >
        
      
      </Text>
    </Text>
    <Text
      style={
        Array [
          null,
          Object {},
        ]
      }
    >
      
    
    </Text>
  </Text>
  <Text
    style={
      Array [
        null,
        null,
      ]
    }
  >
    
    
  </Text>
</View>
`;

exports[`<HTMLView/> should render text in a (nested) <Text/> element 1`] = `
<View>
  <Text
    style={
      Array [
        null,
        null,
      ]
    }
  >
    This is some text
  </Text>
</View>
`;

exports[`<HTMLView/> should render ul bullets 1`] = `
<View>
  <Text
    onLongPress={null}
    onPress={null}
  >
    <Text
      onLongPress={null}
      onPress={null}
      style={null}
    >
      <Text
        style={
          Array [
            null,
            Object {},
          ]
        }
      >
        • 
      </Text>
      <Text
        style={
          Array [
            null,
            Object {},
          ]
        }
      >
         a 
      </Text>
      

    </Text>
    <Text
      onLongPress={null}
      onPress={null}
      style={null}
    >
      <Text
        style={
          Array [
            null,
            Object {},
          ]
        }
      >
        • 
      </Text>
      <Text
        style={
          Array [
            null,
            Object {},
          ]
        }
      >
         b 
      </Text>
    </Text>
  </Text>
</View>
`;
